kvs/redis.go

```diff
@@ -42,7 +44,21 @@ type Redis struct {
 	vk     int
 }
 
-func NewRedis(host, port, pass string, kv, vk int) (*Redis, error) {
+func loopPing(client *redis.Client, numRetry int) error {
```

Comment:
The number of retries depends on the backoff algorithm. 
So I would recommend that you should use latency control
```suggestion
func loopPing(client *redis.Client, retryFreq, waitTime time.Duration) error {
	end := time.NewTicker(waitTime)
	tick := time.NewTicker(retryFreq)

	defer func() {
		end.Stop()
		tick.Stop()
	}()

	var err error
	for {
		select {
		case <-end.C:
			return err
		case <-tick.C:
			_, err = client.Ping().Result()
			if err == nil {
				return nil
			}
		}
	}
}
```

Reply:
@kpango thank you for your nice advice. I followed it.